File: Application/Services/Operations/StockProduct/Dtos/ProductKind/SegmentDto.cs
using System.Collections.Generic;
using Application.Services.Shared.Dtos;

namespace Application.Services.Operations.StockProduct.ProductKind
{
    public class SegmentDto : RootBaseDto
    {
        public string Name { get; set; }
        public int ProductId { get; set; }
        public ProductDto Product { get; set; }
        public List<ManufacturerDto> Manufacturers { get; set; }


    }
}

File: Views/src/components/stock-product/product/dtos/segment-dto.ts
import { RootBase } from "src/shared/entities-dtos/root-base";
import { ManufacturerDto } from "./manufacturer-dto";
import { ProductTypeDto } from "./product-type-dto";
import { ProductDto } from "./product";

    export class SegmentDto  extends RootBase
    {
        name: string;
        productTypeId: number;
        productType: ProductTypeDto;
        manufacturers:ManufacturerDto[];
        products: ProductDto[];
    }